duckdb/datadog_checks/duckdb/check.py

```diff
@@ -0,0 +1,166 @@
+
+
+class DuckdbCheck(AgentCheck):
+
+    # This will be the prefix of every metric and service check the integration sends
+    __NAMESPACE__ = 'duckdb'
+
+    def __init__(self, name, init_config, instances):
+        super(DuckdbCheck, self).__init__(name, init_config, instances)
+
+        self.db_name = self.instance.get('db_name')
+        self.connection_attempt = int(self.instance.get('connection_attempt', 3))
+
+        self.tags = self.instance.get('tags', [])
+        self._connection = None
+        self._connect_params = None
+        self._tags = []
+        self._query_errors = 0
+
+        manager_queries = deepcopy(DEFAULT_QUERIES)
+
+        self._query_manager = QueryManager(
+            self,
+            self._execute_query_raw,
+            queries=manager_queries,
+            tags=self.tags,
+            error_handler=self._executor_error_handler,
+        )
+        self.check_initializations.append(self.initialize_config)
+        self.check_initializations.append(self._query_manager.compile_queries)
+
+    def check(self, _):
+        retry_delay = 5
+        max_retries = self.connection_attempt
+        for attempt in range(1, max_retries + 1):
+            try:
+                with self.connect() as conn:
+                    if conn:
+                        self._connection = conn
+                        self._query_manager.execute()
+                        break
+            except Exception as e:
+                self.log.warning('Unable to connect to the database:  "%s" , retrying...', e)
+                if attempt < max_retries:
+                    time.sleep(retry_delay)
+                else:
+                    self.log.error('Max connection retries reached')
+
+    def _execute_query_raw(self, query):
+        with closing(self._connection.cursor()) as cursor:
+            query = query.format(self.db_name)
+            curs = cursor.execute(query)
+            if len(curs.fetchall()) < 1:
+                self._query_errors += 1
+                self.log.warning('Failed to fetch records from query: `%s`.', query)
+                return None
+            for row in cursor.execute(query).fetchall():
+                pattern_version = r"\bversion\b"
+                query_version = re.search(pattern_version, query)
+                if query_version:
+                    query_name = 'version'
+                else:
+                    # Try to find the field name from the query
+                    pattern = r"(?i)\bname\s*=\s*'([^']+)'"
+                    query_name = re.search(pattern, query).group(1)
+                try:
+                    yield self._queries_processor(row, query_name)
+                except Exception as e:
+                    self.log.debug(
+                        'Unable to process row returned from query "%s", skipping row %s. %s', query_name, row, e
+                    )
+                    yield row
+
+    def _queries_processor(self, row, query_name):
```

Comment:
Perhaps this function could be inlined inside _execute_query_raw and the branches could be removed. It would make it clear that query_name is just used for debug logging.